addons/mail/models/mail_message.py

```diff
@@ -1027,7 +1027,7 @@ def _message_format(self, format_reply=True, msg_vals=None, for_current_user=Fal
                 displayed_tracking_ids = allowed_tracking_ids
                 if record and hasattr(record, '_track_filter_for_display'):
                     displayed_tracking_ids = record._track_filter_for_display(displayed_tracking_ids)
-                vals["needaction_partner_ids"] = notifs.filtered(lambda n: not n.is_read).res_partner_id.ids
+                vals["needaction"] = self.env.user.partner_id in notifs.filtered(lambda n: not n.is_read).res_partner_id
```

Comment:
extra `notifs` variable is no longer necessary, as it basically does 2 filtering in a row on the same thing, just checking the presence of the partner is enough

This should always be False for public user, code previously handled that in JS but you removed it.